From 06f7081e7a5b714c22d100a0bd354ad71c1facef Mon Sep 17 00:00:00 2001
From: Ramneek Mehresh <ramneek.mehresh@freescale.com>
Date: Thu, 30 Oct 2014 10:54:08 +0530
Subject: [PATCH 056/666] drivers:usb:pm: Fix pm defect in rcpm driver for usb
 ip

Pass device-tree usb node offset to each usb device structure.
This is required by rcpm driver to get mask setting for
ip exception register to make/remove usb as wake-up source
for sleep/deep-sleep

Signed-off-by: Ramneek Mehresh <ramneek.mehresh@freescale.com>
Change-Id: I4e6b0f931fb29a6311b36a97255d994c2ce0ad05
Reviewed-on: http://git.am.freescale.net:8181/22478
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Nikhil Badola <nikhil.badola@freescale.com>
Reviewed-by: Matthew Weigel <Matthew.Weigel@freescale.com>
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/usb/core/usb.c           | 1 +
 drivers/usb/host/fsl-mph-dr-of.c | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/drivers/usb/core/usb.c b/drivers/usb/core/usb.c
index 62e1906..061dd35 100644
--- a/drivers/usb/core/usb.c
+++ b/drivers/usb/core/usb.c
@@ -592,6 +592,7 @@ struct usb_device *usb_alloc_dev(struct usb_device *parent,
 	dev->dev.dma_mask = bus->sysdev->dma_mask;
 	dev->dev.dma_pfn_offset = bus->sysdev->dma_pfn_offset;
 	set_dev_node(&dev->dev, dev_to_node(bus->sysdev));
+	dev->dev.of_node = bus->controller->of_node;
 	dev->state = USB_STATE_ATTACHED;
 	dev->lpm_disable_count = 1;
 	atomic_set(&dev->urbnum, 0);
diff --git a/drivers/usb/host/fsl-mph-dr-of.c b/drivers/usb/host/fsl-mph-dr-of.c
index cb7964b..475fb149 100644
--- a/drivers/usb/host/fsl-mph-dr-of.c
+++ b/drivers/usb/host/fsl-mph-dr-of.c
@@ -103,6 +103,8 @@ static struct platform_device *fsl_usb2_device_register(
 	else
 		dma_set_mask(&pdev->dev, DMA_BIT_MASK(32));
 
+	pdev->dev.of_node = ofdev->dev.of_node;
+
 	retval = platform_device_add_data(pdev, pdata, sizeof(*pdata));
 	if (retval)
 		goto error;
-- 
2.7.4

